features/filmStrip/components/VideoThumbnail.js

```diff
@@ -107,26 +120,46 @@ class VideoThumbnail extends Component {
             <VideoThumbnailContainer
                 focused={this.props.participant.focused}
                 onClick={this._onClick}>
-                {streams.video && <Video
-                    stream={streams.video}
-                    onPlaying={this._onVideoPlaying}/>}
-                {streams.audio && <Audio
-                    stream={streams.audio}/>}
+
+                {streams.audio &&
+                    <Audio stream={streams.audio}/>}
+
+                {streams.video && !this.props.videoMuted &&
+                    <Video
+                        stream={streams.video}
+                        onPlaying={this._onVideoPlaying}/>}
+
+                {this.props.participant.role === PARTICIPANT_ROLE.MODERATOR &&
+                    <ModeratorIndicator />}
+
+                {this.props.participant.speaking &&
+                    <DominantSpeakerIndicator />}
+
+                {this.props.audioMuted &&
+                    <AudioMutedIndicator />}
+
+                {this.props.videoMuted &&
+                    <VideoMutedIndicator />}
+
             </VideoThumbnailContainer>
         );
     }
 }
 
+
 /**
  * VideoThumbnail component's property types.
  *
  * @static
  */
 VideoThumbnail.propTypes = {
+    audioMuted: React.PropTypes.bool,
     audioTrack: React.PropTypes.object,
-    dispatch: React.PropTypes.func,
```

Comment:
@legastero, I'm getting eslint errors from your PR. I'll fix them this time but please note that we have 'npm run eslint' now and we should make sure to run it before we create PRs. Thank you!